crates/uv-workspace/src/pyproject.rs

```diff
@@ -504,6 +504,45 @@ pub struct ToolUv {
     )]
     pub environments: Option<SupportedEnvironments>,
 
+    /// A list of required platforms, for packages that lack source distributions.
+    ///
+    /// When a package does not have a source distribution, it's availability will be limited to
+    /// the platforms supported by its built distributions (wheels). For example, if a package only
+    /// publishes wheels for Linux, then it won't be installable on macOS or Windows.
+    ///
+    /// By default, uv requires each package to include at least one wheel that is compatible with
+    /// the designated Python version. The `required-environments` setting can be used to ensure that
+    /// the resulting resolution contains wheels for specific platforms, or fails if no such wheels
+    /// are available.
+    ///
+    /// While the `environments` setting _limits_ the set of environments that uv will consider when
+    /// resolving dependencies, `required-environments` _expands_ the set of platforms that uv _must_
+    /// support when resolving dependencies.
+    ///
+    /// For example, `environments = ["sys_platform == 'darwin'"]` would limit uv to solving for
+    /// macOS (and ignoring Linux and Windows). On the other hand, `required-environments = ["sys_platform == 'darwin'"]`
+    /// would _require_ that any package without a source distribution include a wheel for macOS in
+    /// order to be installable.
```

Comment:
I re-did the docs here. Is it any better, @zanieb @BurntSushi?

Reply:
Beautiful. Thank you!